cmd/argocd/commands/plugin_test.go

```diff
@@ -0,0 +1,156 @@
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// pluginPath refers to the path of the plugin found. For tests, the plugins are in testdata dir
+var pluginPath string
+
+// Test_ArgoCD_Plugin_Successful_Execution tests that a plugin found in the PATH gets executed successfully
+func Test_ArgoCD_Plugin_Successful_Execution(t *testing.T) {
+	pluginHandler := NewDefaultPluginHandler([]string{"argocd"})
+	pluginHandler.lookPath = func(file string) (string, error) {
+		if file == "argocd-foo" {
+			pluginPath = "testdata/argocd-foo"
+		}
+		return pluginPath, nil
+	}
+	cmd := NewCommand()
+	cmd.SilenceErrors = true
+	cmd.SilenceUsage = true
+	args := []string{"argocd", "foo"}
+	cmd.SetArgs(args[1:])
+
+	err := cmd.Execute()
+	require.Error(t, err)
+
+	pluginErr := pluginHandler.HandleCommandExecutionError(err, true, args)
+	assert.Equal(t, "testdata/argocd-foo", pluginPath)
+	require.NoError(t, pluginErr)
+}
+
+// TestNormalArgoCDCommandExecution_WhenPluginWithSameNameIsFound tests that a normal argocd command get executed successfully such as argocd version even if a plugin with the same name (argocd-version) is present in PATH
+func TestNormalArgoCDCommandExecution_WhenPluginWithSameNameIsFound(t *testing.T) {
+	pluginHandler := NewDefaultPluginHandler([]string{"argocd"})
+	pluginHandler.lookPath = func(file string) (string, error) {
+		if file == "argocd-version" {
+			pluginPath = "testdata/argocd-version"
+		}
+		return pluginPath, nil
+	}
+	args := []string{"argocd", "version", "--short", "--client"}
+	buf := new(bytes.Buffer)
+	cmd := NewVersionCmd(&argocdclient.ClientOptions{}, nil)
+	cmd.SetArgs(args[1:])
+	cmd.SetOut(buf)
+	cmd.SilenceErrors = true
+	cmd.SilenceUsage = true
+
+	err := cmd.Execute()
+	require.NoError(t, err)
+	output := buf.String()
+	assert.Equal(t, "argocd: v99.99.99+unknown\n", output)
+}
+
+// Test_Normal_ArgoCD_Command_Error checks for an error that would be received on failure of execution of a normal argocd command
+func Test_Normal_ArgoCD_Command_Error(t *testing.T) {
+	pluginHandler := NewDefaultPluginHandler([]string{"argocd"})
+	pluginHandler.lookPath = func(file string) (string, error) {
+		if file == "argocd-version" {
+			pluginPath = "testdata/argocd-version"
+		}
+		return pluginPath, nil
+	}
+	args := []string{"argocd", "version", "--non-existent-flag"}
+	cmd := NewVersionCmd(&argocdclient.ClientOptions{}, nil)
+	cmd.SetArgs(args[1:])
+	cmd.SilenceErrors = true
+	cmd.SilenceUsage = true
+
+	err := cmd.Execute()
+	require.Error(t, err)
+
+	pluginErr := pluginHandler.HandleCommandExecutionError(err, true, args)
+	// although the lookPath (if executed) will set the pluginPath to testdata/argocd-foo, the plugin path never gets updated because we never execute handlePluginCommand() function
+	// Why? argocd version is a normal argocd command, hence we never try to look for a plugin in this case
+	assert.Equal(t, "", pluginPath)
```

Comment:
```suggestion
	assert.Equal(t, "", pluginPath)
```

This and every other test case gets passed locally but fails here. Looking at the error message
```
=== FAIL: cmd/argocd/commands Test_Normal_ArgoCD_Command_Error (0.00s)
time="2025-01-06T11:13:40Z" level=error msg="Error: unknown flag: --non-existent-flag\n"
    plugin_test.go:83: 
        	Error Trace:	/home/runner/work/argo-cd/argo-cd/cmd/argocd/commands/plugin_test.go:83
        	Error:      	Not equal: 
        	            	expected: ""
        	            	actual  : "testdata/argocd-foo"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-
        	            	+testdata/argocd-foo
        	Test:       	Test_Normal_ArgoCD_Command_Error
```

the actual value is `testdata/argocd-version` but i'm not sure why lookPath func is taking the `argocd-foo` always as file input. cc @blakepettersson